### PR TITLE
[ASCII-1162] Make it easier to add a new dependencies to diagnose

### PR DESCRIPTION
**Pull request by ogaca-dd**
### What does this PR do?

This PR adds a structure to make it easier to add new dependencies to diagnose suite.

### CODE REVIEW

File: comp/core/flare/flare.go [hunk 1]
```diff
@@ -43,24 +44,24 @@ type dependencies struct {
...
+	log          log.Component
+	config       config.Component
+	invAgent     inventoryagent.Component
+	params       Params
+	providers    []types.FlareCallback
+	diagnoseDeps diagnose.SuitesDeps
 }
 
 func newFlare(deps dependencies) (Component, rcclient.TaskListenerProvider, error) {
+	// TODO FIX this uninitialize variable.
```
Comment (ogaca-dd) [inline comp/core/flare/flare.go:56]: This will be fixed in a later PR.

**Review by Kaderinho — APPROVED** [review]

**Review by zhuminyi — APPROVED** [review]
LGTM

**Pull request merged**